### PR TITLE
Clean up recursive function

### PR DESCRIPTION
**Pull request by thomasyu888**
I realized that having repetitive code between Project and Folder creation was inefficient and refactored the function a bit.

### CODE REVIEW

**Review by mfazza — APPROVED** [review]
💯

File: synapseformation/client.py [hunk 1]
```diff
@@ -56,29 +56,34 @@ def _create_synapse_resources(config: dict, creation_cls: SynapseCreation,
...
+    # function is called from within the for loop
+    # Error: entity not specified
+    entity = None
     if isinstance(config, dict) and config.get('type') == "Project":
-        project_ent = creation_cls.get_or_create_project(name=config['name'])
-        _add_acl(creation_cls.syn, project_ent, config.get('acl', []))
-        parent_id = project_ent.id
+        entity = creation_cls.get_or_create_project(name=config['name'])
+    elif isinstance(config, dict) and config.get('type') == "Folder":
+        print(config)
```
Comment (thomasyu888) [inline synapseformation/client.py]:
```suggestion
```

File: synapseformation/client.py [hunk 2]
```diff
@@ -56,29 +56,33 @@ def _create_synapse_resources(config: dict, creation_cls: SynapseCreation,
...
-        parent_id = project_ent.id
+        entity = creation_cls.get_or_create_project(name=config['name'])
+    elif isinstance(config, dict) and config.get('type') == "Folder":
+        entity = creation_cls.get_or_create_folder(
+            name=config['name'], parentId=parentid
+        )
+    else:
+        # Loop through folders and create them
+        for folder_config in config:
+            # Must pull out children if it exists
```
Comment (thomasyu888) [inline synapseformation/client.py]:
```suggestion
```

File: synapseformation/client.py [hunk 3]
```diff
@@ -56,29 +56,32 @@ def _create_synapse_resources(config: dict, creation_cls: SynapseCreation,
...
+                                      parentid=parentid)
+    if entity is not None:
+        parent_id = entity.id
         config['id'] = parent_id
-        # Get children if exists
+        # Get ACL if exists
+        config.get('acl', [])
+        _add_acl(creation_cls.syn, entity, config.get('acl', []))
+        # Get children if exists, but don't run recursive function
+        # if no children exists
```
Comment (thomasyu888) [inline synapseformation/client.py]:
```suggestion
```

File: synapseformation/client.py [hunk 4]
```diff
@@ -49,31 +49,31 @@ def _create_synapse_resources(config: dict, creation_cls: SynapseCreation,
...
+        # Loop through folders and create them
+        for folder_config in config:
+            _create_synapse_resources(folder_config, creation_cls,
+                                      parentid=parentid)
+    if entity is not None:
+        parent_id = entity.id
         config['id'] = parent_id
-        # Get children if exists
+        # Get ACL if exists
+        config.get('acl', [])
```
Comment (thomasyu888) [inline synapseformation/client.py]:
```suggestion
```

**Pull request merged**